cmd/containerd-shim-runhcs-v1/exec_hcs.go

```diff
@@ -49,14 +49,20 @@ func newHcsExec(
 	id, bundle string,
 	isWCOW bool,
 	spec *specs.Process,
-	io hcsoci.UpstreamIO) shimExec {
+	io hcsoci.UpstreamIO,
+	isTemplate bool) shimExec {
 	log.G(ctx).WithFields(logrus.Fields{
 		"tid":    tid,
 		"eid":    id, // Init exec ID is always same as Task ID
 		"bundle": bundle,
 		"wcow":   isWCOW,
 	}).Debug("newHcsExec")
 
+	if tid != id && isTemplate {
```

Comment:
Can we error out on this condition before calling this? I don't think the expectation from calling newHcsExec is to ever get a nil exec back but I may be missing something in these changes that makes this ok.

Reply:
Good catch. I just wanted to make sure that we don't accidentally call `newHcsExec` with isTemplate set to true but the exec is not the init process. However, currently there are only two places from where we call this and both have been correctly updated with this change. So I will remove this condition.